tests/bluetooth/blob_io_flash/src/main.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/bluetooth/mesh.h>
+
+#include "mesh/blob.h"
+
+#define SLOT1_PARTITION		slot1_partition
+#define SLOT1_PARTITION_ID	FIXED_PARTITION_ID(SLOT1_PARTITION)
+#define SLOT1_PARTITION_SIZE	FIXED_PARTITION_SIZE(SLOT1_PARTITION)
+
+static struct bt_mesh_blob_io_flash blob_flash_stream;
+
+static inline size_t chunk_size(const struct bt_mesh_blob_block *block,
+				uint16_t chunk_idx)
+{
+	if ((chunk_idx == block->chunk_count - 1) &&
+	    (block->size % BLOB_CHUNK_SIZE_MAX(BT_MESH_TX_SDU_MAX))) {
+		return block->size % BLOB_CHUNK_SIZE_MAX(BT_MESH_TX_SDU_MAX);
+	}
+
+	return BLOB_CHUNK_SIZE_MAX(BT_MESH_TX_SDU_MAX);
+}
+
+static uint8_t block_size_to_log(size_t size)
+{
+	uint8_t block_size_log = 0;
+
+	while (size > 1) {
+		size = size / 2;
+		block_size_log++;
+	}
+
+	return block_size_log;
+}
+
+ZTEST_SUITE(blob_io_flash, NULL, NULL, NULL, NULL, NULL);
+
+ZTEST(blob_io_flash, test_chunk_read)
+{
+	const struct flash_area *fa = NULL;
+	struct bt_mesh_blob_xfer xfer;
+	struct bt_mesh_blob_block block = { 0 };
+	struct bt_mesh_blob_chunk chunk = { 0 };
+	size_t remaining = SLOT1_PARTITION_SIZE;
+	off_t block_idx = 0;
+	uint16_t chunk_idx = 0;
+	uint8_t chunk_data[BLOB_CHUNK_SIZE_MAX(BT_MESH_TX_SDU_MAX)];
+	uint8_t test_data[SLOT1_PARTITION_SIZE];
+	uint8_t ctrl_data[SLOT1_PARTITION_SIZE];
+	size_t tests_data_offset = 0;
+	int i, err;
+
+	/* Fill test data with pattern */
+	for (i = 0; i < SLOT1_PARTITION_SIZE; i++) {
+		test_data[i] = i % 0xFF;
+	}
+
+	err = flash_area_open(SLOT1_PARTITION_ID, &fa);
+	zassert_equal(err, 0, "Preparing test data failed with err=%d", err);
+
+	err = flash_area_erase(fa, 0, ARRAY_SIZE(ctrl_data));
+	zassert_equal(err, 0, "Preparing test data failed with err=%d", err);
+
+	err = flash_area_write(fa, 0, test_data, ARRAY_SIZE(ctrl_data));
+	zassert_equal(err, 0, "Preparing test data failed with err=%d", err);
+
+	err = flash_area_read(fa, 0, ctrl_data, ARRAY_SIZE(ctrl_data));
+	zassert_equal(err, 0, "Preparing test data failed with err=%d", err);
+
+	zassert_mem_equal(ctrl_data, test_data, ARRAY_SIZE(ctrl_data),
+			  "Incorrect data written into flash");
+
+	memset(ctrl_data, 0, SLOT1_PARTITION_SIZE);
+
+	err = bt_mesh_blob_io_flash_init(&blob_flash_stream,
+					 SLOT1_PARTITION_ID, 0);
+	zassert_equal(err, 0, "BLOB I/O init failed with err=%d", err);
+	flash_area_close(fa);
```

Comment:
Logically, `flash_area_close(fa);` should be called before `bt_mesh_blob_io_flash_init` since `flash_area_..` functions perform access to flash map. The access should be logically completed. Of course, right now, `flash_area_close` does nothing and really doesn't matter to call it at all. However, if somebody in the future will add any logic to this function(at least function header description assumes that), the test behavior might be broken.

I'd suggest to add `flash_area_close(fa);` before `bt_mesh_blob_io_flash_init` to prevent issues with maintenance.

Reply:
makes sense to me, moved this call as you suggested